File: 프로그래머스/unrated/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.java
// //1.일반 풀이

// class Solution {
//     public int solution(int[][] arr) {
//         for(int i = 0; i < arr.length; i++)
//             for(int j = 0; j < arr.length; j++)
//                 if(arr[i][j]!=arr[j][i]) return 0;
//         return 1;
//     }
// }

// //2.스트림 풀이
import java.util.stream.IntStream;
class Solution {
    public int solution(int[][] arr) {
        return IntStream.range(0, arr.length).allMatch(i -> IntStream.range(0,arr[0].length).map(j->arr[i][j]-arr[j][i]).allMatch(k -> k == 0)) ? 1: 0;
    }
}

File: 프로그래머스/0/181831. 특별한 이차원 배열 2/특별한 이차원 배열 2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> arr) {
    for(int i = 0; i < arr.size() / 2; i++){
        for(int j = i + 1; j < arr.size(); j++){
            if(arr[i][j] != arr[j][i]) return 0;
        }
    }
    return 1;
}